micrograd.py

```diff
@@ -96,9 +96,7 @@ def _backward():
         out._backward = _backward
         return out
 
-def topological_sort(value: Value, sorted_list: Optional[list] = None) -> list[Value]:
-    if sorted_list is None:
-        sorted_list = []
+def topological_sort(value: Value, sorted_list: list = []) -> list[Value]:
```

Comment:
mutable default params in Python are an antipattern
https://google.github.io/eng-practices/review/reviewer/